src/java.base/share/classes/java/lang/foreign/SegmentAllocator.java

```diff
@@ -733,8 +733,14 @@ private MemorySegment allocateNoInit(MemoryLayout layout) {
 
     @ForceInline
     private MemorySegment allocateNoInit(MemoryLayout layout, long size) {
+        long byteSize;
+        try {
```

Comment:
I don't think we want this. While it is more correct, that would negatively affect performance. If there's an overflow, let it go negative, it will be caught anyway

Reply:
True. But we will get the wrong type of exception.

Reply:
We could change the exception type to `ArithmeticException` in this case and get performance and correctness at the same time.

Reply:
I simply don't think we should be using `multiplyExact` here.